clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1125,7 +1125,7 @@ public void commitSync() {
     public void commitSync(final Map<TopicPartition, OffsetAndMetadata> offsets) {
         acquire();
         try {
-            coordinator.commitOffsetsSync(new HashMap<>(offsets), Long.MAX_VALUE);
```

Comment:
Making a copy is important as we can run into deadlocks if the user synchronizes on the received map. Someone actually reported one such issue.